org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -423,6 +439,12 @@ private synchronized void doNodeInitialisation() {
         coordinatorHandler.serializeNetwork();
     }
 
+    private synchronized void reinitializeDevice() {
```

Comment:
Reading this now, the method name seems wrong. This method reinitializes the thing handler, and not the device. In fact, it will only reinitialize the device if the `THING_PROPERTY_DEVICE_INITIALIZED` was set to `false` before (which you do in `handleConfigurationUpdate`).

I suggest to either rename the method, or set the thing property in here, to fix this issue.

Reply:
OK. I will move the statement
```
getThing().setProperty(ZigBeeBindingConstants.THING_PROPERTY_DEVICE_INITIALIZED, Boolean.FALSE.toString());
```
into this method. I think then it's clear what the method is made for.